packages/Breadcrumbs/src/Breadcrumbs.js

```diff
@@ -15,6 +15,10 @@ function Breadcrumbs(props) {
   const hasOnlyOneChild = React.Children.count(children) === 1;
   const [isCollapsed, setIsCollapsed] = React.useState(shouldShowExpandButton);
 
+  React.useLayoutEffect(() => {
+    setIsCollapsed(shouldShowExpandButton);
+  }, [children, shouldShowExpandButton]);
```

Comment:
🤔 wouldn't this re-render always if you add children? if a parent component change their state this will get trigger again, isn't it?